app/controllers/contacts_controller.rb

```diff
@@ -0,0 +1,20 @@
+class ContactsController < ApplicationController
+
+  def create
+    @company = find_company(params[:company_id])
+    contact = @company.contacts.new(contacts_params)
+    if contact.save
+      flash[:success] = "#{contact.name} was added as a contact to #{@company.name}"
+      redirect_to company_path(@company)
+    else
+      flash[:danger] = "Contact must have a Name and Email to be valid."
+      redirect_to company_path(@company)
```

Comment:
if we're redirecting to the same place regardless of if the contact saves or not, let's put the redirect line after the conditional so we don't have to repeat ourselves.